feat/fix-storeMenu,order

jvn1o opened the pull request:
## 📌 개요
- 전체적으로 추가 및 수정

## 🔨 변경 사항
- UserRole이 Manager, Master인 경우 softDelete 처리 된 메뉴들도 조회할 수 있게 로직 추가, 기존처럼 Owner는 본인의 StoreMenu는 softDelete 되지 않은 메뉴들만 조회할 수 있다.
- phoneNumber의 데이터 정책에 따른 수정
- address의 최소 10자 ~ 최대 255자 제약 조건에 따른 Test 수정
- StoreMenu: 목록 조회 시 Manager, Master 모두 삭제된 메뉴들 안 보이도록 변경
- Order: 고객이 주문 생성 후 5분 이내 && 요청 상태가 주문 중 일 때에만 취소 가능하도록 설계
- Order: 고객이 주문 수락 행위를 하려고 하면 실패하는 테스트 케이스 추가
- Order: OrderController에 Swagger 누락된 부분 추가

## ✅ 테스트
- [x] 단위/통합 테스트 통과 확인
- [x] 로컬 실행 후 정상 동작 확인

## ☑️ 체크리스트
- [x] 코드 컨벤션 준수
- [x] 문서화 (ERD, API 명세서 등) 반영

## 🙋 리뷰 요청사항(선택)
- (예시)이런 A라는 문제가 있는데, B와 C중에 어떤게 괜찮을까요?

jvn1o committed:
feat/fix-storeMenu-API

UserRole이 Manager, Master인 경우 softDelete 처리 된 메뉴들도 조회할 수 있게 로직 추가, 기존처럼 Owner는 본인의 StoreMenu는 softDelete 되지 않은 메뉴들만 조회할 수 있다.

jvn1o committed:
final Complete

- phoneNumber의 데이터 정책에 따른 수정

- address의 최소 10자 ~ 최대 255자 제약 조건에 따른 Test 수정

jvn1o committed:
feat/fix-storeMenu-API

Manager, Master 모두 삭제된 메뉴들 안 보이도록 정책 결정

jvn1o committed:
feat/fix-order-api

- 고객이 주문 생성 후 5분 이내 && 요청 상태가 주문 중 일 때에만 취소 가능하도록 설계
- 고객이 주문 수락 행위를 하려고 하면 실패하는 테스트 케이스 추가
- OrderController에 Swagger 누락된 부분 추가

skfkgla commented on src/main/java/com/sparta/delivery/backend/order/dto/ResCheckOutOrderDto.java:
```diff
@@ -28,7 +28,7 @@ public class ResCheckOutOrderDto {
 	private String addressDetail;
 
-	@Schema(description = "고객 연락처", example = "010-1234-5678")
+	@Schema(description = "고객 연락처", example = "01012345678")
```
응답은 010-1234-5678
이렇게 가는게 맞습니다!!

요청시에만 숫자만 들어가있는지 체크하시면 됩니다!!

skfkgla commented on src/main/java/com/sparta/delivery/backend/store/menu/service/StoreMenuService.java:
```diff
@@ -90,8 +91,20 @@ public Page<ResGetListStoreMenuDto> getStoreMenusByStoreId(
+		// 기본값은 삭제되지 않은 값들만 조회
+		storeMenuList = storeMenuRepository.findAllByStoreIdAndDeletedAtIsNull(storeId, pageable);
+
+		/* 현재는 Manager, Master 도 삭제된 값들을 보지 못하도록 설계하기로 결정
```
이건 지워주셔도 좋을 것 같습니다!

jvn1o committed:
feat/fix-storeMenu-test

고객이나 다른 가게의 점주는 숨김 처리된 메뉴를 조회하지 않고, 해당 가게의 점주, 매니저, 마스터만 숨김 처리 (삭제 x) 된 메뉴까지 전부 조회가 가능하다.

skfkgla commented on src/main/java/com/sparta/delivery/backend/store/menu/service/StoreMenuService.java:
```diff
@@ -86,15 +86,21 @@ public Page<ResGetListStoreMenuDto> getStoreMenusByStoreId(
-		// 기본값은 삭제되지 않은 값들만 조회
-		storeMenuList = storeMenuRepository.findAllByStoreIdAndDeletedAtIsNull(storeId, pageable);
+		// Customer || Owner 이지만 본인의 가게가 아닌 경우
+		if (user.getRole() == UserRoleEnum.CUSTOMER || !store.getOwner().getUser().getPublicId().equals(user.getPublicId())) {
```
내부에서는 publicId 사용하기 보단 Long값인 id를 사용하는게 좋을 것 같습니다!
그리고 해당 store는 fetchJoin이 아니어서, N+1 문제가 발생할 것 같습니다.
store의 Owner와 User는 fetchJoin으로 가져와 주세요!!

jvn1o committed:
feat/fix-order-dto

요청은 하이픈 있도록
